docs/pages/task/explain.md

```diff
@@ -6,84 +6,362 @@ In this scenario the question is in format of RPG code and the output is an expl
+- Implement error handling to manage cases where no matching record is found. This could involve returning a specific value or message indicating that no match was found, improving the program's robustness and user feedback.
+- Use a keyed read operation instead of reading through the entire file to improve performance. This approach directly accesses the record with the matching key, reducing the time complexity from linear to constant time for the lookup. 
+``` 
+
+### sum_output
+The sum_output part offers a business summary in a couple of sentences, explaining the overall purpose of the program
+
+`sum_output.md` has the content for `program.pgm.rpgle`:
+
+```text
+### Summary 
+The program is designed to retrieve the account number (`PAccNo`) associated with a given user ID (`P_UserId`) from a file (`AccPf`). It iterates through the records in the file, checking each record to find a match for the user ID. Once a match is found, the corresponding account number is returned
 ```
 
-and `metadata.txt` has
+### Notes
+
+- Use `rpgleforILERPG` code blocks and `rpgforOPMRPG` code blocks.
+- Use `###` for first-level headings and `####` for sub-level headings.
+- Avoid using `**` for bold text use backticks ` ` to highlight keywords or important terms.
+- Use the term `fully-free` instead of `full free format`.
+- Use `ILE RPG` instead of `RPGLE`.
+- Do not expand `RPG` as `Report Program Generator`. Always refer to it as `RPG`. 
+
+### metadata.txt
+
+Complete compilable ILE RPG source file for programs or modules or copybooks
 
 ```yaml
-difficulty: 0
-language: rpg4ff
-scope: file
-use: train
+difficulty: <rating>
+language: <language>
+scope: <scope>
+use: <usage>
 ```
 
-Where
+Structure for Procedure or Subroutine
 
-- `difficulty` - the difficulty of the explanation as rated up to 5.  Since this is trivial we rate it as 0.
-- `language` - the language of the snippet of code being explained in this case `rpg4ff` which is RPG IV fully free
-- `scope` - the scope of the language being explained in this case `file` i.e. the source file
-- `use` - `train` means the data is used for training the LLM, the alternative would be `eval` if this was being used to evaluate an LLM
+```yaml
+source: <source_member_name>
+start: <start_line_number>
+end: <end_line_number>
+difficulty: <rating>
+language: <language>
+scope: <scope>
+use: <usage>
+```
+- `difficulty`: The difficulty of the explanation as rated up to 5. This helps to understand the complexity of the code being documented. For example, a trivial explanation might be rated as 0.
+- `language`: The language of the snippet of code being explained. For example, `rpg4ff` indicates RPG IV fully free format. If the source file contains both fixed format and free format RPGLE, it should be marked as `rpg4fc`.
```

Comment:
Append reference to a another page that describes all of the languages in metadata.txt